src/main/resources/proto/googleidentity/oauth2/request/OAuth2Request.proto

```diff
@@ -40,11 +40,26 @@ message OAuth2Request {
     RequestAuth request_auth = 1;
 
     message RequestBody{
+
+        enum ResponseType {
+             EMPTY = 0;
+             CODE = 1;
+             TOKEN = 2;
+        }
+
         // Request response type
-        string response_type = 1;
+        ResponseType response_type = 1;
+
+        enum GrantType {
+            NONE = 0;
+            AUTHORIZATION_CODE = 1;
+            IMPLICIT = 2;
+            REFRESH_TOKEN = 3;
+            JWT_ASSERTION = 4;
+        }
```

Comment:
Use a common enum between ClientDetails and OAuth2Request protos. I'd recommend creating a file to contain shared enums that both protos import.